api_types.ts

```diff
@@ -828,6 +828,23 @@ export enum PermissionSyncMode {
   PermissionAware = 'PermissionAware',
 }
 
+/**
+ * The type of content being indexed, which determines how the property is processed and queried.
+ * Messaging: Chat or instant messaging content
+ * Document: General document content
+ * Email: Email message content
+ * Comment: User comments or feedback
+ * Help: Help documentation or support content
+ * @hidden
+ */
+export enum IndexCategory {
```

Comment:
Nit, I might call this `ContentCategory` (and ditto for the property) to be a little bit more flexible in how this can be used in the future. "index" is a little implementation-specific. Generally in the SDK we like to frame things as "describe your data abstractly, and we'll use that info in ever-changing ways in the platform to make best use of your pack".